gems/manageiq-providers-vmware-vcd/app/models/manageiq/providers/vmware_vcd/cloud_manager/refresher.rb

```diff
@@ -0,0 +1,71 @@
+class ManageIQ::Providers::VmwareVcd::CloudManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
+  include ::EmsRefresh::Refreshers::EmsRefresherMixin
+
+  def collect_inventory_for_targets(ems, targets)
+    vcd = ems.vcd_service
+
+    inv = {
+      :orgs  => [],
+      :vdcs  => [],
+      :vapps => [],
+      :vms   => [],
+    }
+
+    vcd.organizations.all.each do |org|
+      inv[:orgs].push(org)
+    end
+
+    inv[:orgs].each do |org|
+      org.vdcs.all.each do |vdc|
+        inv[:vdcs].push(vdc)
+      end
```

Comment:
You are right.

there are two types of customers:
- those that connect to a vCD provider (i.e. a cloud provider), and then only have access to this API
- those that have also access to the underlying infra provider.

They are independent but will be connected. VMware sees vApps as resource groups, and vCD sees ID of VMware